### PR TITLE
VOTE-577: improve date validation

### PR DESCRIPTION
**Pull request by mlloydbixal**
[VOTE-577](https://cm-jira.usa.gov/browse/VOTE-577)

Birthday field validates each field as you move through them but still uses only one error message. The HTML5 messages are now hidden but maintain functionality. Labels close correctly after label text. All error states have been removed and replaced with class changing functionality.

### CODE REVIEW

File: src/components/FormSections/PersonalInfo.jsx [hunk 1]
```diff
@@ -253,25 +299,26 @@ function PersonalInfo(props){
...
                 </Fieldset>
                 </div>
             </Grid>
             )}
 
             {telephoneFieldState && (
                 <Grid tablet={{ col: 5 }}>
-                    <div className={(parseInt(telephoneFieldState.required) && handleErrors.phone_number) ? 'error-container' : ''}>
-                        <Label className="text-bold" htmlFor="phone-number">{phoneNumberField.label}{(telephoneFieldState.required === "1") && <span className='required-text'>*</span>}</Label>
+                    <div className="input-parent">
```
Comment (rayestrada) [inline src/components/FormSections/PersonalInfo.jsx:318]: onblur error triggering on phone field even when not required.

File: src/components/FormSections/PersonalInfo.jsx [hunk 2]
```diff
@@ -185,11 +228,12 @@ function PersonalInfo(props){
                             autoComplete="off"
                             required={parseInt(dobFieldState.required)}
                             data-testid="dateInputGroup"
-                            onBlur={event => { if (!event.currentTarget.contains(event.relatedTarget)) checkDateValues(); }}
-                        >
+                            onBlur={e => { if (!e.currentTarget.contains(e.relatedTarget)) toggleError(e, checkDateValues('all')) }}
```
Comment (rayestrada) [inline src/components/FormSections/PersonalInfo.jsx:231]: I am able to proceed with with a future date of birth `01 01 2025` even though an error displays.
Screenshot 2024-02-05 at 4 16 39 PM

File: src/components/FormSections/PersonalInfo.jsx [hunk 3]
```diff
@@ -321,16 +370,20 @@ function PersonalInfo(props){
...
                 </div>
             </Grid>
 
             {raceFieldState && (
                 <Grid row gap>
                     <Grid tablet={{ col: 4 }}>
-                        <div className={((parseInt(raceFieldState.required)) && handleErrors.race) ? 'error-container' : ''}>
-                            <Label className="text-bold" htmlFor="race-ethic-group-select">{raceField.label}{(raceFieldState.required === "1") && <span className='required-text'>*</span>}
+                        <div className="input-parent">
+                            <Label className="text-bold" htmlFor="race-ethic-group-select">
```
Comment (rayestrada) [inline src/components/FormSections/PersonalInfo.jsx:384]: race ethnic field displaying error onblur even if it's not required.
Screenshot 2024-02-05 at 4 19 19 PM

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: src/components/FormSections/PersonalInfo.jsx (hunk 1 above)
Comment (mlloydbixal) [reply src/components/FormSections/PersonalInfo.jsx:320]: fixed with pattern requirement using regex to trigger form invalid

File: src/components/FormSections/PersonalInfo.jsx (hunk 2 above)
Comment (mlloydbixal) [reply src/components/FormSections/PersonalInfo.jsx:234]: removed custom year validation and html5 pattern

**Review by rayestrada — APPROVED** [review]

**Pull request merged**